python/packages/litellm/agent_framework_litellm/_chat_client.py

```diff
@@ -0,0 +1,178 @@
+                "model_id parameter, or LITE_LLM_MODEL_ID env variable must be provided for LiteLlmChatClient"
+            )
+
+        self.model_id = str(lite_llm_settings.model_id)
+
+        super().__init__(api_base=api_base, api_key=api_key, model_id=self.model_id, client=None, **kwargs)  # type: ignore
+
+    def make_completion_streaming_request(self, **options_dict: Any) -> CustomStreamWrapper:
+        options_dict["model"] = options_dict.pop("model_id")
+        lite_llm_response = completion(stream=True, **options_dict)
+        # Completion conditionally returns this depending on streaming vs non-streaming
+        return cast(CustomStreamWrapper, lite_llm_response)
+
+    def make_completion_nonstreaming_request(self, **options_dict: Any) -> ModelResponse:
+        options_dict["model"] = options_dict.pop("model_id")
+        lite_llm_response = completion(stream=False, **options_dict)
+        # Completion conditionally returns this depending on streaming vs non-streaming
+        return cast(ModelResponse, lite_llm_response)
+
+    def lite_llm_to_openai_completion(self, response: ModelResponse) -> ChatCompletion:
+        # Convert a LiteLLM ResponsesAPIResponse to an OpenAI Response. LiteLLM aims to match the OpenAI A  I,
+        # however in the future there may be differences that need to be accounted for here.
+        return cast(ChatCompletion, response)
+
+    def lite_llm_event_to_openai_event(self, event: ModelResponseStream) -> ChatCompletionChunk:
+        # Convert a LiteLLM ResponsesAPIResponse to an OpenAI Response. LiteLLM aims to match the OpenAI API,
+        # however in the future there may be differences that need to be accounted for here.
+        openai_event = cast(ChatCompletionChunk, event)
+
+        # LiteLLM does not providet this as a first-class field, so we map it here.
+        openai_event.usage = event.get("usage", None)
+        return openai_event
+
+    def lite_llm_to_openai_response(self, response: ModelResponse) -> ChatCompletion:
+        """Convert a LiteLLM ModelResponse to an OpenAI ChatCompletion."""
+        # OpenAI parsing code currently directly checks for OpenAI classes. However,
+        # LiteLLM implements the OpenAI API via its own classes, compatable via
+        # duck typing. Therefore, we need to convert the LiteLLM ModelResponse
+        # to an OpenAI ChatCompletion for compatibility.
+
+        # You may need to adapt this mapping based on the actual structure of ModelResponse
+        # and the expected fields of ChatCompletion.
+        # Here is a basic example assuming similar structure:
+        return ChatCompletion(**response.dict())
+
+    async def _inner_get_response(
+        self,
+        *,
+        messages: MutableSequence[ChatMessage],
+        chat_options: ChatOptions,
+        **kwargs: Any,
+    ) -> ChatResponse:
+        options_dict = self._prepare_options(messages, chat_options)
+        options_dict["model_id"] = self.model_id
```

Comment:
this should already be done in the prepare_options, and people should be able to set the model at runtime.